src/library_fs.js

```diff
@@ -403,29 +403,39 @@ FS.staticInit();` +
         FS.FSStream = /** @constructor */ function() {
           this.shared = { };
         };
-        FS.FSStream.prototype = {
+        FS.FSStream.prototype = {};
+        Object.defineProperties(FS.FSStream.prototype, {
           object: {
+            /** @this {FS.FSStream} */
             get: function() { return this.node; },
+            /** @this {FS.FSStream} */
             set: function(val) { this.node = val; }
           },
           isRead: {
+            /** @this {FS.FSStream} */
             get: function() { return (this.flags & {{{ cDefine('O_ACCMODE') }}}) !== {{{ cDefine('O_WRONLY') }}}; }
           },
           isWrite: {
+            /** @this {FS.FSStream} */
             get: function() { return (this.flags & {{{ cDefine('O_ACCMODE') }}}) !== {{{ cDefine('O_RDONLY') }}}; }
           },
           isAppend: {
+            /** @this {FS.FSStream} */
             get: function() { return (this.flags & {{{ cDefine('O_APPEND') }}}); }
           },
           flags: {
+            /** @this {FS.FSStream} */
             get: function() { return this.shared.flags; },
+            /** @this {FS.FSStream} */
             set: function(val) { this.shared.flags = val; },
           },
           position : {
-            get function() { return this.shared.position; },
+            /** @this {FS.FSStream} */
+            get: function() { return this.shared.position; },
```

Comment:
There was a missing `:` here.

Reply:
If this missing colon is enough to fix the issue maybe we don't need to change anything else?

Reply:
It is not enough to fix the issue.

Reply:
To define getters/setters, you have to use `defineProperties`. What this does is it makes `position` into an object, so `stream.position.set` is a function. If you say `stream.position = 0` that overwrites it with a number. So the streams don't share their position.